dynawo/sources/API/PAR/PARParametersSetImpl.h

```diff
@@ -235,7 +236,7 @@ class ParametersSet::Impl : public ParametersSet {
 
   std::string id_; /**< Parameters' set id */
   std::map<std::string, boost::shared_ptr<Parameter> > parameters_; /**< Map of the parameters */
-  std::map<std::string, boost::shared_ptr<Reference> > references_; /**< Map of the references */
+  boost::unordered_map<std::string, boost::shared_ptr<Reference> > references_; /**< Map of the references */
```

Comment:
Here you make the choice to convert the map to unordered_map (I know you like it ;)) but then you need to do twice a first for loop to put the elements in an order way (and I think this is the only uses of this object). Is the passage to unordered_map efficient in this situation?

Reply:
My motivation (beyond that I love hashmaps ;)) : those manual ordering are used only once whereas the lookup into this structure is used for all parameters in  setParameterFromPARFile

Reply:
The other solution is to separate references from par parameters at some point and iterate first on one and then and the other one. I can do this if you prefer, or we can keep it as plan B if we see performances issues with this change.

Reply:
You are the boss ;)
More seriously, I just wanted to double check with you so the current implementation is fine for me.